ddl/src/main/java/com/adaptivescale/rosetta/ddl/targets/bigquery/BigQueryDDLGenerator.java

```diff
@@ -42,118 +49,100 @@ public String createColumn(Column column) {
-                        .append("`.");
-            }
-            builder.append("`").append(table.getName()).append("`; \n");
-        }
-
-        builder.append("CREATE TABLE ");
-
-        if (table.getSchema() != null && !table.getSchema().isBlank()) {
-            builder.append("`")
-                    .append(table.getSchema())
-                    .append("`.");
+            stringBuilder.append(dropTable(table));
         }
 
-        builder.append("`")
-                .append(table.getName())
-                .append("`")
-                .append("(")
-                .append(definitionAsString)
-                .append(");");
+        createParams.put("schemaName", table.getSchema());
+        createParams.put("tableName", table.getName());
+        createParams.put("tableCode", definitionAsString);
+        stringBuilder.append(TemplateEngine.process(TABLE_CREATE_TEMPLATE, createParams));
 
-        return builder.toString();
+        return stringBuilder.toString();
     }
 
     @Override
     public String createTableSchema(Table table) {
-        StringBuilder stringBuilder = new StringBuilder();
-        if (table.getSchema() != null && !table.getSchema().isBlank()) {
-            stringBuilder
-                    .append("CREATE SCHEMA IF NOT EXISTS ")
-                    .append(table.getSchema())
-                    .append(";");
-        }
-        return stringBuilder.toString();
+        Map<String, Object> params = new HashMap<>();
+        params.put("schemaName", table.getSchema());
+        return TemplateEngine.process(SCHEMA_CREATE_TEMPLATE, params);
     }
 
-    @Override
     public String createDatabase(Database database, boolean dropTableIfExists) {
         StringBuilder stringBuilder = new StringBuilder();
 
-        Set<String> schemas = database.getTables().stream().map(Table::getSchema)
-                .filter(s -> s != null && !s.isEmpty()).collect(Collectors.toSet());
+        Set<String> schemas = database.getTables().stream().map(Table::getSchema).filter(s -> s != null && !s.isEmpty()).collect(Collectors.toSet());
         if (!schemas.isEmpty()) {
             stringBuilder.append(
                     schemas
                             .stream()
-                            .map(schema -> "CREATE SCHEMA IF NOT EXISTS " + schema)
-                            .collect(Collectors.joining(";\r\r"))
-
+                            .map(this::createSchema)
+                            .collect(Collectors.joining())
             );
-            stringBuilder.append(";\r");
+            stringBuilder.append("\r");
         }
 
         stringBuilder.append(database.getTables()
                 .stream()
                 .map(table -> createTable(table, dropTableIfExists))
-                .collect(Collectors.joining("\r")));
+                .collect(Collectors.joining("\r\r")));
 
         return stringBuilder.toString();
     }
 
     @Override
     public String alterColumn(ColumnChange change) {
```

Comment:
In the previous version it looks like we supported the process of DROP NOT NULL as well. Can we add that back or there was a reason why this is not included here?